### PR TITLE
fs: vectored IO API for filesystem read/write

### PR DESCRIPTION
**Pull request by Ralith**
#1103 with a windows error fixed

### CODE REVIEW

**Comment by saghul** [comment]
Jenkinds doesn't seem to have the Windows slave available, I'll run the test suite later today. Thanks!

File: src/win/fs.c [hunk 1]
```diff
@@ -1588,13 +1632,27 @@ int uv_fs_read(uv_loop_t* loop, uv_fs_t* req, uv_file fd, void* buf,
 }
 
 
-int uv_fs_write(uv_loop_t* loop, uv_fs_t* req, uv_file fd, const void* buf,
-    size_t length, int64_t offset, uv_fs_cb cb) {
+int uv_fs_write(uv_loop_t* loop,
+                uv_fs_t* req,
+                uv_file file,
```
Comment (saghul) [inline src/win/fs.c]: uv_file fd.

**Comment by saghul** [comment]
Fails to compile on windows because there are 4 places where req->length is still used. Lines 542, 606, 1120 and 1983.

**Comment by Ralith** [comment]
Applied style tweak and removed or replaced references to req->length as appropriate.

**Comment by saghul** [comment]
Compiles and test suite passes on Windows now, thanks @Ralith! Landed in https://github.com/joyent/libuv/commit/13dd3502a7aedf4284559a88b3662e222723b7ce

**Pull request closed without merging**